src/coreclr/jit/lower.cpp

```diff
@@ -3916,6 +3916,17 @@ bool Lowering::TryLowerConditionToFlagsNode(GenTree* parent, GenTree* condition,
             BlockRange().Remove(relop);
             BlockRange().Remove(relopOp2);
         }
+#ifdef TARGET_ARM64
+        else if (optimizing && relop->OperIs(GT_EQ, GT_NE, GT_LT, GT_LE, GT_GE, GT_GT) &&
+                 (IsContainableUnaryOrBinaryOp(relop, relop->gtGetOp1()) ||
+                  IsContainableUnaryOrBinaryOp(relop, relop->gtGetOp2())))
+        {
+            ContainCheckCompare(relop);
+            relop->SetOper(GT_CMP);
+            relop->gtType = TYP_VOID;
+            relop->gtFlags |= GTF_SET_FLAGS;
+        }
+#endif // TARGET_ARM64
```

Comment:
This seems a bit odd. It does exactly what would happen in the `else` case below, unconditionally, except it also does the `ContainCheckCompare`, and also, it doesn't move the `relop` down, as specified in the function contract.

Don't we call ContainCheckCompare in LowerCompare() before we get here?

Reply:
`LowerCompare` does call `ContainCheckCompare` and `LowerCompare` should have already been called by now. I will check this.

Reply:
Ok, it turns out I didn't even need to have that code there. 😒